aws-lc-rs/build.rs

```diff
@@ -2,28 +2,74 @@
 // SPDX-License-Identifier: Apache-2.0 OR ISC
 
 fn main() {
-    let mutually_exclusives_count = [cfg!(feature = "non-fips"), cfg!(feature = "fips")]
-        .iter()
-        .filter(|x| **x)
-        .count();
-
-    if mutually_exclusives_count > 1 {
-        eprint!("fips and non-fips are mutually exclusive crate features.");
-        std::process::exit(1);
-    }
+    let has_mutually_exclusive_features = cfg!(feature = "non-fips") && cfg!(feature = "fips");
+    assert!(
+        !has_mutually_exclusive_features,
+        "`fips` and `non-fips` are mutually exclusive crate features."
+    );
 
     // This appears asymmetric, but it reflects the `cfg` statements in lib.rs that
     // require `aws-lc-sys` to be present when "fips" is not enabled.
-    let at_least_one_count = [cfg!(feature = "aws-lc-sys"), cfg!(feature = "fips")]
-        .iter()
-        .filter(|x| **x)
-        .count();
-
-    if at_least_one_count < 1 {
-        eprint!(
-            "one of the following features must be specified: \
-        aws-lc-sys, non-fips, or fips."
+    // if `fips` is enabled, then use that
+    let sys_crate = if cfg!(feature = "fips") {
+        "aws-lc-fips-sys"
+    } else if cfg!(feature = "aws-lc-sys") {
+        "aws-lc-sys"
+    } else {
+        panic!(
```

Comment:
One problem consumers have is that our build requires the consumer to specify which sys crate to use. Instead, I think it would be better [automatically select](
https://doc.rust-lang.org/cargo/reference/build-scripts.html#rustc-cfg) the "aws-lc-sys" when neither was explicitly selected.
```
   let sys_crate = if cfg!(feature = "aws-lc-fips-sys") {
        "aws-lc-fips-sys"
    } else if cfg!(feature = "aws-lc-sys") {
        "aws-lc-sys"
    } else {
        println!("cargo:rustc-cfg=feature=\"aws-lc-sys\"]");
        "aws-lc-sys"
    };
```

Reply:
Hmm I'm not sure that'll actually work:

> Note that this does not affect Cargo’s dependency resolution. This cannot be used to enable an optional dependency, or enable other Cargo features.